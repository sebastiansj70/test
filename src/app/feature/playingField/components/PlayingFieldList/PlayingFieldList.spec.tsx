import '@testing-library/jest-dom';
import React from 'react';
import { RenderResult, render, wait, fireEvent } from '@testing-library/react';
import { PlayingFieldList } from '.';
import { setTextEvent } from 'app/shared/utils/test';

describe('PlayingFieldList test', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof PlayingFieldList>;

  beforeEach(() => {
    componentProps = {
      playingFieldLists: [
        {
          idCancha: 1,
          statusCancha: 'Bueno',
        },
        {
          idCancha: 2,
          statusCancha: 'Malo',
        },
      ],
      playingField: {
        idCancha: 3,
        statusCancha: 'Regular',
      },
      updatePlayingField: jest.fn(),
      savePalyingField: jest.fn(),
    };
    componentWrapper = render(<PlayingFieldList {...componentProps} />);
  });

  it('should match snapshot', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });

  it('Deber tener un titulo', () => {
    const elem = componentWrapper.container;
    const tittle = elem.querySelectorAll('h1');

    expect(tittle[0].textContent).toMatch(/Lista de campos/i);
  });
});
