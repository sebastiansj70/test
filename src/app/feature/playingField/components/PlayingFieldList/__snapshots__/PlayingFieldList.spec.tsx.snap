// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayingFieldList test should match snapshot 1`] = `
<div>
  <div
    class="sc-AxirZ dZWKyK"
  >
    <h1
      class="sc-AxiKw cRbuiW"
    >
      Lista de campos de juego
    </h1>
    <div
      id="canchaList"
    >
      <div
        class="sc-AxjAm jPIHtA"
      >
        <h3>
          Número de Cancha : 1
        </h3>
        <h3>
          Estado de Cancha : Bueno
        </h3>
      </div>
    </div>
    <div
      id="canchaList"
    >
      <div
        class="sc-AxjAm jPIHtA"
      >
        <h3>
          Número de Cancha : 2
        </h3>
        <h3>
          Estado de Cancha : Malo
        </h3>
      </div>
    </div>
    <div />
  </div>
</div>
`;
